plutus-core/plutus-core/src/PlutusCore/Constant/Meaning.hs

```diff
@@ -94,7 +94,7 @@ data BuiltinRuntime term =
 
 -- | A 'BuiltinRuntime' for each builtin from a set of builtins.
 newtype BuiltinsRuntime fun term = BuiltinsRuntime
-    { unBuiltinRuntime :: Array fun (BuiltinRuntime term)
+    { unBuiltinRuntime :: Array Int (BuiltinRuntime term)
```

Comment:
There isn't an optimized version of `Array` for `Int`? Maybe it gets specialized anyway.

Reply:
Zachary just helped me out in slack and yeah, [there is](https://hackage.haskell.org/package/primitive-0.7.1.0/docs/Data-Primitive-Array.html).

However we're trying to measure two things here:

1. how much of an improvement it is to store builtins as `Int`s
2. how much of an improvement it is to use a "specialized" version of `Array`

and it makes sense to measure them separately. Particularly given the fact that the indexing function in the `primitive` library does not do bounds checking and we do a shit ton of bounds checking, which is something we could drop without changing the arrays library.

Reply:
Yep, fine.